src/main/java/org/candlepin/model/EnvironmentCurator.java

```diff
@@ -71,19 +70,61 @@ public List<Consumer> getEnvironmentConsumers(String environmentId) {
             .getResultList();
     }
 
-    public CandlepinQuery<Environment> listForOwner(Owner o) {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria()
-            .add(Restrictions.eq("owner", o));
+    /**
+     * Retrieves a list on environments based on the criteria and given types
+     * @param owner
+     *  The owner of the environments
+     * @param envName
+     *  An exact name for the environments
+     * @param type
+     *  A list of types to filter the environments
+     * @return
+     *  A List of environments based on the criteria
+     */
+    public List<Environment> listByType(@NotNull Owner owner, String envName, List<String> type) {
```

Comment:
The `@NotNull` annotation doesn't inherently do any validation -- it needs to be checked with a validator. Both of which are pretty heavy for doing input validation here (not to mention the silliness that is annotation-based almost-meta programming), so we're better off with a traditional null check and IAE/fast exit.